### PR TITLE
Refactor based on changes in dns_common

### PR DESCRIPTION
**Pull request by ananace**
Also adds support for `AAAA` and `CNAME` records.

For reference: theforeman/smart-proxy#17906

### CODE REVIEW

File: test/infoblox_test.rb [hunk 1]
```diff
@@ -20,7 +20,7 @@ def test_create_a
     ip = '10.1.1.1'
 
     @provider.expects(:a_record_conflicts).with(fqdn, ip).returns(-1)
```
Comment (dmitri-d) [inline test/infoblox_test.rb]: Here and in the rest of the tests below: the super class takes care of checks for conflicts for us and has tests to verify these checks are called (https://github.com/theforeman/smart-proxy/blob/develop/test/dns_common/record_test.rb#L174). All infoblox provider tests need to verify is that when do_create is called, an ib_create_*/ib_delete_* appropriate for the record type is called. These should limit each test to a single expectation.

**Comment by dmitri-d** [comment]
Looks good to me, other than the comment re: tests. More changes around dns provider interface coming up though (https://github.com/theforeman/smart-proxy/pull/511, https://github.com/theforeman/smart-proxy/pull/507) which may require additional changes in infoblox provider. Not sure if should wait or deal with them as they come (probably the latter?).

**Comment by ananace** [comment]
Neither of the PRs seem to actually affect the code beneath dns_common, only naming conventions and de-duplication inside of the checks.
Though I guess a small update to follow the consistent naming might be in the future.

**Comment by ananace** [comment]
So, has there been any progress in mergeability checks?

We're looking to move away from our internal (ugly, hacked) python proxy for IB, but we have IPv6 in place so would also like not losing that feature.

**Comment by dmitri-d** [comment]
Could you add tests for creation and removal of 'AAAA' records? Otherwise looks good.

**Comment by ananace** [comment]
Hopefully these are enough tests for that?

**Comment by dmitri-d** [comment]
Merged, thanks @ace13!

**Pull request merged**